SignalServiceKit/src/Network/API/Requests/OWSRequestFactory.m

```diff
@@ -26,6 +31,256 @@ + (TSRequest *)disable2FARequest
     return [TSRequest requestWithUrl:[NSURL URLWithString:textSecure2FAAPI] method:@"DELETE" parameters:@{}];
 }
 
++ (TSRequest *)acknowledgeMessageDeliveryRequestWithSource:(NSString *)source timestamp:(UInt64)timestamp
+{
+    OWSAssert(source.length > 0);
+    OWSAssert(timestamp > 0);
+
+    NSString *path = [NSString stringWithFormat:@"v1/messages/%@/%llu", source, timestamp];
+
+    return [TSRequest requestWithUrl:[NSURL URLWithString:path] method:@"DELETE" parameters:@{}];
+}
+
++ (TSRequest *)deleteDeviceRequestWithDevice:(OWSDevice *)device
+{
+    OWSAssert(device);
+
+    NSString *path = [NSString stringWithFormat:textSecureDevicesAPIFormat, @(device.deviceId)];
```

Comment:
How would you feel about moving path fragments into here from TSConstants? 

No biggie, just a small annoyance when adding new requests that you have to add something to that file.

Reply:
Some of the newer requests do that.  It probably made more sense to constant-ize, when we might have separate requests that would PUT/GET/DELETE the same path.  Anyway, let's move the paths ... but not in this PR.